DataFormats/CaloRecHit/interface/CaloRecHit.h

```diff
@@ -4,27 +4,52 @@
 #include "DataFormats/DetId/interface/DetId.h"
 #include <ostream>
 
+#ifdef __CUDA_ARCH__
+  __constant__
+#else
+  constexpr
+#endif
+  uint32_t calo_rechit_masks[] = {
+    0x00000000u,0x00000001u,0x00000003u,0x00000007u,0x0000000fu,0x0000001fu,
+    0x0000003fu,0x0000007fu,0x000000ffu,0x000001ffu,0x000003ffu,0x000007ffu,
+    0x00000fffu,0x00001fffu,0x00003fffu,0x00007fffu,0x0000ffffu,0x0001ffffu,
+    0x0003ffffu,0x0007ffffu,0x000fffffu,0x001fffffu,0x003fffffu,0x007fffffu,
+    0x00ffffffu,0x01ffffffu,0x03ffffffu,0x07ffffffu,0x0fffffffu,0x1fffffffu,
+    0x3fffffffu,0x7fffffffu,0xffffffffu};
 
 /** \class CaloRecHit
  * 
  *\author J. Mans - Minnesota
  */
 class CaloRecHit {
 public:
-  CaloRecHit(); // for persistence
-  explicit CaloRecHit(const DetId& id, float energy, float time, uint32_t flags = 0, uint32_t aux=0);
+  constexpr CaloRecHit() : energy_(0), time_(0), flags_(0), aux_(0) {}
+  constexpr explicit CaloRecHit(const DetId& id, float energy, float time, 
```

Comment:
While at it, maybe drop the `explicit` as it has no effect here?

Reply:
done

Reply:
> While at it, maybe drop the `explicit` as it has no effect here?

Actually it has an effect, but I don't think we ever use that.
With `explicit` you can construct a `CaloRecHit` with any of these syntaxes:
```c++
CaloRecHit crh1{ id, energy, time };
CaloRecHit crh2( id, energy, time );
CaloRecHit crh3 = CaloRecHit{ id, energy, time };
CaloRecHit crh4 = CaloRecHit( id, energy, time );
```
(although these last two are better written with `auto`:
```c++
auto crh3 = CaloRecHit{ id, energy, time };
auto crh4 = CaloRecHit( id, energy, time );
```
).

Without `explicit` you could use also this:
```c++
CaloRecHit crh5 = { id, energy, time };
```

As I said, not something we are likely to use anyway.

Reply:
See more details on [Stakoverflow](https://stackoverflow.com/a/39122237/2050986).

Reply:
Thanks @fwyzard, I stand corrected (I had missed this detail of C++11). Then I don't have an opinion whether `explicit` should be kept here or not.